File: hackerrank/strings/anagram.js
const areAnagrams =
    (a, b) => {
      if (a.length != b.length) {
        return false;
      }

      const char_q_a = new Map();
      const char_q_b = new Map();

      for (const char of a) {
        char_q_a.set(char, (char_q_a.get(char) || 0) + 1);
      };

      for (const char of b) {
        char_q_b.set(char, (char_q_b.get(char) || 0) + 1);
      }

      if (char_q_a.size !== char_q_b.size) {
        return false;
      }

      for (const [char, count] of char_q_a) {
        if (char_q_b.get(char) !== count) {
          return false;
        }
      }
      return true;
    }

              // Usage
              console.log(`amagrams = ${areAnagrams('anagram', 'margana')}`)

File: hackerrank/strings/anagram.py
def areAnagrams(a, b):
    if len(a) != len(b):
        return False

    # Frequency character maps and compare
    char_q_a = {}
    char_q_b = {}

    for char in a:
        char_q_a[char] = char_q_a.get(char, 0) + 1

    for char in b:
        char_q_b[char] = char_q_b.get(char, 0) + 1

    return char_q_a == char_q_b

# Usage
print(f"'anagram' and 'margana' {areAnagrams('anagram', 'margana')}")
